beacon_node/http_api/src/lib.rs

```diff
@@ -1807,20 +1834,40 @@ pub fn serve<T: BeaconChainTypes>(
                             .filter(|&att| query_filter(att.data()))
                             .cloned(),
                     );
-                    Ok(api_types::GenericResponse::from(attestations))
+                    let slot = query
+                        .slot
+                        .or_else(|| {
+                            attestations
+                                .first()
+                                .map(|att| att.data().slot)
+                                .or_else(|| chain.slot_clock.now())
+                        })
+                        .ok_or(warp_utils::reject::custom_server_error(
+                            "unable to read slot clock".to_string(),
+                        ))?;
+                    let fork_name = chain.spec.fork_name_at_slot::<T::EthSpec>(slot);
```

Comment:
What if someone queries the pool at the fork boundary? There may be attestations from both forks

Reply:
that's a great point.. looks like we'll have to convert to the new format for all of these attestations

Seems to mean Pawan's idea is the best way to approach it: https://github.com/sigp/lighthouse/pull/5744#discussion_r1672185892